## Vérification les bases de données

In [3]:
import pandas as pd
import numpy as np

#### Fichier 1 : order_item_refunds

In [2]:
# 1.1. chargement des données
df_order_item_refunds = pd.read_csv("../../MODULE 7 - Proje final/Toy store/order_item_refunds.csv")

In [3]:
# 1.2. Informations générales
df_order_item_refunds.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1731 entries, 0 to 1730
Data columns (total 5 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   order_item_refund_id  1731 non-null   int64  
 1   created_at            1731 non-null   object 
 2   order_item_id         1731 non-null   int64  
 3   order_id              1731 non-null   int64  
 4   refund_amount_usd     1731 non-null   float64
dtypes: float64(1), int64(3), object(1)
memory usage: 67.7+ KB


- Le DataFrame contient 1731 lignes (clients),5 colonnes 
| Column | Interprétation rapide |
|:---|:---|
|order_item_refund_id| pas de valeurs manquantes, conserve ce type.|
|created_at| pas de valeurs manquantes, c'est une variable en format datetime, à convertir en datetime.|
|order_item_id| pas de valeurs manquantes, conserve ce type. |
|order_id| pas de valeurs manquantes, conserve ce type. |
|refund_amount_usd| pas de valeurs manquantes,correct pour les montants financiers.|

In [4]:
# 1.3. Vérifier les statistiques descriptives
df_order_item_refunds.describe().round(2)

,order_item_refund_id,order_item_id,order_id,refund_amount_usd
count,1731.00,1731.00,1731.00,1731.00
mean,866.00,18472.21,15868.24,49.30
std,499.84,11438.07,9096.06,4.96
min,1.00,57.00,57.00,29.99
25%,433.50,7417.00,7412.00,49.99
50%,866.00,19858.00,17375.00,49.99
75%,1298.50,26900.00,22539.50,49.99
max,1731.00,39950.00,32255.00,59.99


In [5]:
# 1.4. Vérifier les doublons
print(f"Nombre de doublons : {df_order_item_refunds.duplicated().sum()}")

Nombre de doublons : 0


-  Pas de doublons dans le fichier : order_item_refunds

#### Fichier 2 : order_items

In [6]:
# 2.1. chargement des données
df_order_items = pd.read_csv("../../MODULE 7 - Proje final/Toy store/order_items.csv")

In [7]:
# 2.2. Informations générales
df_order_items.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40025 entries, 0 to 40024
Data columns (total 7 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   order_item_id    40025 non-null  int64  
 1   created_at       40025 non-null  object 
 2   order_id         40025 non-null  int64  
 3   product_id       40025 non-null  int64  
 4   is_primary_item  40025 non-null  int64  
 5   price_usd        40025 non-null  float64
 6   cogs_usd         40025 non-null  float64
dtypes: float64(2), int64(4), object(1)
memory usage: 2.1+ MB


- Le DataFrame contient 40025 lignes (clients),7 colonnes 
| Column | Interprétation rapide |
|:---|:---|
|order_item_id| pas de valeurs manquantes, conserve ce type.|
|created_at| pas de valeurs manquantes, c'est une variable en format datetime, à convertir en datetime.|
|order_id| pas de valeurs manquantes, conserve ce type. |
|product_id| pas de valeurs manquantes, conserve ce type. |
|is_primary_item| pas de valeurs manquantes,à convertir en booléen (bool).|
|price_usd | pas de valeurs manquantes,conserve ce type.|
|cogs_usd | pas de valeurs manquantes,conserve ce type.|    

In [8]:
# 2.3. Vérifier les statistiques descriptives
df_order_items.describe().round(2)

,order_item_id,order_id,product_id,is_primary_item,price_usd,cogs_usd
count,40025.00,40025.00,40025.00,40025.00,40025.00,40025.00
mean,20013.00,17121.96,1.77,0.81,48.43,18.05
std,11554.37,9053.77,1.09,0.39,8.01,3.86
min,1.00,1.00,1.00,0.00,29.99,9.49
25%,10007.00,9871.00,1.00,1.00,49.99,19.49
50%,20013.00,17490.00,1.00,1.00,49.99,19.49
75%,30019.00,24818.00,2.00,1.00,49.99,19.49
max,40025.00,32313.00,4.00,1.00,59.99,22.49


In [9]:
# 2.4. Vérifier les doublons
print(f"Nombre de doublons : {df_order_items.duplicated().sum()}")

Nombre de doublons : 0


-  Pas de doublons dans le fichier : order_items

#### Fichier 3 : orders

In [10]:
# 3.1. chargement des données
df_orders = pd.read_csv("../../MODULE 7 - Proje final/Toy store/orders.csv")

In [11]:
# 3.2. Informations générales
df_orders.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32313 entries, 0 to 32312
Data columns (total 8 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   order_id            32313 non-null  int64  
 1   created_at          32313 non-null  object 
 2   website_session_id  32313 non-null  int64  
 3   user_id             32313 non-null  int64  
 4   primary_product_id  32313 non-null  int64  
 5   items_purchased     32313 non-null  int64  
 6   price_usd           32313 non-null  float64
 7   cogs_usd            32313 non-null  float64
dtypes: float64(2), int64(5), object(1)
memory usage: 2.0+ MB


- Le DataFrame contient 32313 lignes (clients),8 colonnes 
| Column | Interprétation rapide |
|:---|:---|
|order_id| pas de valeurs manquantes, conserve ce type.|
|created_at| pas de valeurs manquantes, c'est une variable en format datetime, à convertir en datetime.|
|website_session_id| pas de valeurs manquantes, conserve ce type. |
|user_id| pas de valeurs manquantes, conserve ce type. |
|primary_product_id| pas de valeurs manquantes,conserve ce type.|
|items_purchased| pas de valeurs manquantes,conserve ce type.|
|price_usd | pas de valeurs manquantes,conserve ce type.|  
|cogs_usd | pas de valeurs manquantes,conserve ce type.|    

In [12]:
# 3.3. Vérifier les statistiques descriptives
df_orders.describe().round(2)

,order_id,website_session_id,user_id,primary_product_id,items_purchased,price_usd,cogs_usd
count,32313.0,32313.00,32313.00,32313.00,32313.00,32313.00,32313.00
mean,16157.0,258292.29,215691.62,1.39,1.24,59.99,22.36
std,9328.1,132427.65,108402.20,0.73,0.43,17.81,6.24
min,1.0,20.00,13.00,1.00,1.00,29.99,9.49
25%,8079.0,144828.00,124135.00,1.00,1.00,49.99,19.49
50%,16157.0,263554.00,221461.00,1.00,1.00,49.99,19.49
75%,24235.0,374799.00,310542.00,2.00,1.00,59.99,22.49
max,32313.0,472818.00,394273.00,4.00,2.00,109.98,41.98


In [13]:
# 3.4. Vérifier les doublons
print(f"Nombre de doublons : {df_orders.duplicated().sum()}")

Nombre de doublons : 0


-  Pas de doublons dans le fichier : orders

In [14]:
#### Fichier 4 : products

In [15]:
# 4.1. chargement des données
df_products = pd.read_csv("../../MODULE 7 - Proje final/Toy store/products.csv")

In [16]:
# 4.2. Informations générales
df_products.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4 entries, 0 to 3
Data columns (total 3 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   product_id    4 non-null      int64 
 1   created_at    4 non-null      object
 2   product_name  4 non-null      object
dtypes: int64(1), object(2)
memory usage: 228.0+ bytes


- Le DataFrame contient 4 lignes (clients),3 colonnes 
| Column | Interprétation rapide |
|:---|:---|
|product_id | pas de valeurs manquantes, conserve ce type.|
|created_at| pas de valeurs manquantes, c'est une variable en format datetime, à convertir en datetime.|
|product_name| pas de valeurs manquantes, conserve ce type. |

In [17]:
# 4.3. Vérifier les statistiques descriptives
df_products.describe().round(2)

,product_id
count,4.00
mean,2.50
std,1.29
min,1.00
25%,1.75
50%,2.50
75%,3.25
max,4.00


In [18]:
# 4.4. Vérifier les doublons
print(f"Nombre de doublons : {df_products.duplicated().sum()}")

Nombre de doublons : 0


-  Pas de doublons dans le fichier : products

#### Fichier 5 : website_pageviews

In [19]:
# 5.1. chargement des données
df_pageviews = pd.read_csv('../../MODULE 7 - Proje final/Toy store/website_pageviews.csv')

In [20]:
# 5.2. Informations générales
df_pageviews.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1188124 entries, 0 to 1188123
Data columns (total 4 columns):
 #   Column               Non-Null Count    Dtype 
---  ------               --------------    ----- 
 0   website_pageview_id  1188124 non-null  int64 
 1   created_at           1188124 non-null  object
 2   website_session_id   1188124 non-null  int64 
 3   pageview_url         1188124 non-null  object
dtypes: int64(2), object(2)
memory usage: 36.3+ MB


- Le DataFrame contient 1188124 lignes (clients),4 colonnes 
| Column | Interprétation rapide |
|:---|:---|
|website_pageview_id| pas de valeurs manquantes, conserve ce type.|
|created_at| pas de valeurs manquantes, c'est une variable en format datetime, à convertir en datetime.|
|website_session_id| pas de valeurs manquantes, conserve ce type. |
|pageview_url| pas de valeurs manquantes, conserve ce type. |

In [21]:
# 5.3. Vérifier les statistiques descriptives
df_pageviews.describe().round(2)

,website_pageview_id,website_session_id
count,1188124.00,1188124.00
mean,594062.50,244458.52
std,342982.00,135619.86
min,1.00,1.00
25%,297031.75,127786.00
50%,594062.50,247808.00
75%,891093.25,362739.00
max,1188124.00,472871.00


In [22]:
# 5.4. Vérifier les doublons
print(f"Nombre de doublons : {df_pageviews.duplicated().sum()}")

Nombre de doublons : 0


-  Pas de doublons dans le fichier : website_pageviews

#### Fichier 6 : website_sessions

In [23]:
# 6.1. chargement des données
df_sessions = pd.read_csv('../../MODULE 7 - Proje final/Toy store/website_sessions.csv')

In [24]:
# 6.2. Informations générales
df_sessions.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 472871 entries, 0 to 472870
Data columns (total 9 columns):
 #   Column              Non-Null Count   Dtype 
---  ------              --------------   ----- 
 0   website_session_id  472871 non-null  int64 
 1   created_at          472871 non-null  object
 2   user_id             472871 non-null  int64 
 3   is_repeat_session   472871 non-null  int64 
 4   utm_source          389543 non-null  object
 5   utm_campaign        389543 non-null  object
 6   utm_content         389543 non-null  object
 7   device_type         472871 non-null  object
 8   http_referer        432954 non-null  object
dtypes: int64(3), object(6)
memory usage: 32.5+ MB


- Le DataFrame contient 472871 lignes (clients),9 colonnes 
| Column | Interprétation rapide |
|:---|:---|
|website_session_id| pas de valeurs manquantes, conserve ce type.|
|created_at| pas de valeurs manquantes, c'une variable en format datetime, à convertir en datetime.|
|user_id| pas de valeurs manquantes, conserve ce type. |
|is_repeat_session| pas de valeurs manquantes,à convertir en booléen (bool).|
|utm_source| pas de valeurs manquantes, conserve ce type.|    
|utm_campaign| pas de valeurs manquantes, conserve ce type.|   
|utm_content| pas de valeurs manquantes, conserve ce type.| 
|device_type | pas de valeurs manquantes, conserve ce type.|  
|http_referer| pas de valeurs manquantes, conserve ce type.|    

In [25]:
# 6.3. Vérifier les statistiques descriptives
df_sessions.describe().round(2)

,website_session_id,user_id,is_repeat_session
count,472871.00,472871.00,472871.00
mean,236436.00,198037.97,0.17
std,136506.24,111993.00,0.37
min,1.00,1.00,0.00
25%,118218.50,101966.50,0.00
50%,236436.00,199483.00,0.00
75%,354653.50,294433.00,0.00
max,472871.00,394318.00,1.00


In [26]:
# 6.4. Vérifier les doublons
print(f"Nombre de doublons : {df_sessions.duplicated().sum()}")

Nombre de doublons : 0


-  Pas de doublons dans le fichier : website_sessions

### Conclusion:

- Aucune valeur manquante dans les fichiers
- Aucun doublon dans les fichiers
- À traiter
| Champs | A traiter |
|:---|:---|
|created_at| à convertir en datetime dans chaque fichier.|
|is_primary_item| pas de valeurs manquantes,à convertir en booléen (bool).|
|is_repeat_session| pas de valeurs manquantes,à convertir en booléen (bool).|
  